### ISSUE TITLE
Make slides automatically

### ISSUE THREAD

**Issue by gruelingpine185** [maaslalani/slides#324]
**Is your feature request related to a problem? Please describe.**
I'm using Slides to render my Advent of Code prompt as well as the README.md files in that repo. I curl the prompt, convert it to markdown, and render it, but often times the content exceeds the slide's boundaries. This means I have to scroll up to see the rest of the document.

What it looks like upon launch:

<img width="647" alt="Screenshot 2024-12-02 at 11 20 24 AM" src="https://github.com/user-attachments/assets/e46626a2-bc1d-43d2-9480-9249e5ccaa4e">

What it looks like after scrolling to the top. Eventually, the output gets muddied by that of previous runs or other commands.

<img width="632" alt="Screenshot 2024-12-02 at 11 21 38 AM" src="https://github.com/user-attachments/assets/f6b79e91-45f4-4957-8375-30ad2e9f0a8b">

<img width="622" alt="Screenshot 2024-12-02 at 11 22 45 AM" src="https://github.com/user-attachments/assets/4a5605a6-d724-4ac8-ab09-a2beaf30d9b4">

**Describe the solution you'd like**
What if Slides broke the document into several slides automatically to fit in view if no slide delimiter `---` was encountered? Or if the nearest slide delimiter is not within the viewing bounds?

**Describe alternatives you've considered**
I tried to write a script that will insert a slide separator every N paragraphs, but that failed miserably. Not to mention, it does not account for long paragraphs or any edge cases that may come up.

As far as using other programs, I have not considered that. Plus, if there is an opportunity to make the program I am using better in some way or another, I'd try that long before switching.

**Comment by tartley** [comment]
It's possible you just want a terminal markdown renderer, not a slideshow. Maybe [glow](https://github.com/charmbracelet/glow) ?

**Comment by gruelingpine185** [comment]
Yeah... looks like what I should have used. Paging would be nice. Thanks
